frog/imports/ui/StudentView/ReactiveHOC.jsx

```diff
@@ -1,15 +1,34 @@
+// @flow
+
 import React, { Component } from 'react';
 import { connection } from '../App/index';
 import generateReactiveFn from '../../api/generateReactiveFn';
 
-const getDisplayName = WrappedComponent =>
-  WrappedComponent.displayName || WrappedComponent.name || 'Component';
+const getDisplayName = (WrappedComponent: any): string => {
+  if (typeof WrappedComponent.displayName === 'string') {
+    return WrappedComponent.displayName;
+  } else if (typeof WrappedComponent.name === 'string') {
+    return WrappedComponent.name;
+  } else {
+    return 'Component';
+  }
+};
 
-const ReactiveHOC = (dataStructure, docId) => WrappedComponent => {
+const ReactiveHOC = (dataStructure: any, docId: string) => (
+  WrappedComponent: Class<Component<*, *, *>>
+) => {
```

Comment:
I had a weird flow error: 
```
Expected React component instead of React$Component
```

I found a fix here, but I will need to read it more carefully to understand it
https://github.com/facebook/flow/issues/4017
https://flow.org/en/docs/types/utilities/#toc-class